ci: investigate linux-race failures in GitHub actions

Currently, the `linux-race` job is consistently failing in GitHub actions (with exit code 143 - SIGTERM).

For example, a [CI run with verbose logging enabled](https://github.com/cockroachdb/pebble/actions/runs/3594961123/jobs/6053873177) fails with the following:

```console
2022-12-01T17:39:21.3663768Z make: *** [Makefile:22: test] Terminated
2022-12-01T17:39:21.5137635Z ##[error]Process completed with exit code 143.
2022-12-01T17:39:21.5192954Z ##[error]The runner has received a shutdown signal. This can happen when the runner service is stopped, or a manually started runner is canceled.
2022-12-01T17:39:21.7086950Z Cleaning up orphan processes
```

Initial indications point at 7d9a5b2e29d53d4296535185c61d46f4a3a82022, though the same failure mode has been observed on branches _without_ this commit (see [here](https://github.com/cockroachdb/pebble/actions/runs/3586007155/jobs/6036268236)).

Pure speculation: we're crossing some kind of GH Action resource limit when running `race`, which results in the platform sending a SIGTERM to the `make` process, which fails the test.

See [this internal thread](https://cockroachlabs.slack.com/archives/C01RV1QS88N/p1669773046452389) for more context.

--- a/sstable/value_block.go
+++ b/sstable/value_block.go
@@ -460,9 +460,12 @@ func (w *valueBlockWriter) addValue(v []byte) (valueHandle, error) {
 	}
 	blockLen = int(vh.offsetInBlock + vh.valueLen)
 	if cap(w.buf.b) < blockLen {
-		size := w.blockSize + w.blockSize/2
-		if size < blockLen {
-			size = blockLen + blockLen/2
+		size := 2 * cap(w.buf.b)
+		if size < 1024 {
+			size = 1024
+		}
+		for size < blockLen {
+			size *= 2
 		}
 		buf := make([]byte, blockLen, size)
 		_ = copy(buf, w.buf.b)

